### PR TITLE
feat: add `prepend_` and `append_` support for config

### PR DESCRIPTION
**Pull request by NSPC911**
woah, am i copying more things from yazi? no way

---

by submitting this pull request, i agree that

- [x] i have run `poe check` to check for any style issues and fixed them
- [x] i have tested rovr (and also ran `poe test` if applicable) to make sure my changes do not break anything
- [x] cache, logs, dotfiles and/or others were not accidentally added to git's tracking history
- [x] my commits (or at least the pr title) follow the conventional commits format as much as possible
- [x] the documentation has been updated wherever necessary (run `poe gen-schema` and `poe gen-keys` if applicable)

## Summary by Sourcery

Add support for prepend_/append_ list modifiers in config merging and clean up right-click menu schema typing and docs.

New Features:
- Support special prepend_*/append_* list keys in config merging to allow extending existing list settings without fully overriding them.

Enhancements:
- Refactor right-click menu item typing to use shared _RightClickItem/_RightClickItemOptionsItem definitions and update references accordingly.
- Move deep_merge functionality from utils into the config module and remove the unused implementation in utils.

Documentation:
- Clarify the right_click settings schema in the developer reference, including that it fully replaces the default list and updating item naming and metadata.

Tests:
- Add tests covering deep_merge behavior, including prepend_/append_ modifiers, nested usage, and type edge cases, as well as tests for natural_size output formats.

## Release Notes

* **Documentation**
  * Updated right-click/context-menu documentation and the generated config schema to better reflect `right_click` item structure, including menu extension guidance (full replacement vs prepend/append).

* **New Features**
  * Added `prepend_` / `append_` semantics for config list keys to control how default lists are extended.

* **Tests**
  * Expanded unit tests for config merging and formatting utilities (including natural size output).

### CODE REVIEW

File: docs/src/content/docs/dev/reference/schema.mdx [hunk 1]
```diff
@@ -841,7 +841,7 @@ Must be one of:
...
 | ---------------------------------------------------------- | --------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | [use_recycle_bin](#settings_use_recycle_bin)               | boolean         | When deleting a file, allow moving the file to the recycle bin.                                                                                                                                                         |
-| [right_click](#settings_right_click)                       | array of object | Configuration for the right-click context menu                                                                                                                                                                          |
+| [right_click](#settings_right_click)                       | array           | Configuration for the right-click context menu. Fully replaces the default list.                                                                                                                                        |
```
[the remark this answers is not in the document]
Comment (NSPC911) [reply docs/src/content/docs/dev/reference/schema.mdx:844]: auto generated, cant do anything about this
